### ISSUE TITLE
[2.2.2] libexpat won't build under Mac OS X 10.4

### ISSUE THREAD

**Issue by jduerstock** [libexpat/libexpat#95]
I realize I'm using an operating system from the crypt, but... what's the recommended way to fix this?
```console
[... expat]$ uname -a
Darwin ibook-g4.local 8.11.0 Darwin Kernel Version 8.11.0: Wed Oct 10 18:26:00 PDT 2007; root:xnu-792.24.17~1/RELEASE_PPC Power Macintosh powerpc
[... expat]$ make
/bin/sh ./libtool --verbose --mode=compile gcc -I./lib -I. -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -fno-strict-aliasing  -DHAVE_EXPAT_CONFIG_H -o lib/xmlparse.lo -c lib/xmlparse.c
libtool: compile:  gcc -I./lib -I. -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -fno-strict-aliasing -DHAVE_EXPAT_CONFIG_H -c lib/xmlparse.c  -fno-common -DPIC -o lib/.libs/xmlparse.o
lib/xmlparse.c:56:3: error: #error You do not have support for any sources of high quality entropy enabled. For end user security, that is probably not what you want. Your options include: * Linux + glibc >=2.25 (getrandom): HAVE_GETRANDOM, * Linux + glibc <2.25 (syscall SYS_getrandom): HAVE_SYSCALL_GETRANDOM, * BSD / macOS (arc4random_buf): HAVE_ARC4RANDOM_BUF, * libbsd (arc4random_buf): HAVE_ARC4RANDOM_BUF + HAVE_LIBBSD, * Windows (RtlGenRandom): _WIN32. If insist on not using any of these, bypass this error by defining XML_POOR_ENTROPY; you have been warned. For CMake, one way to pass the define is: cmake -DCMAKE_C_FLAGS="-pipe -O2 -DHAVE_SYSCALL_GETRANDOM" . If you have reasons to patch this detection code away or need changes to the build system, please open a bug. Thank you!
 # error  \
   ^~~~~
make: *** [lib/xmlparse.lo] Error 1
```

**Comment by hartwork** [comment]
looks like the same issue as #81. We have pull request #84 merged to Git `master` to fix that by now. Could you give Git `master` a try?

**Comment by jduerstock** [comment]
That seemed to fix it, except for this:

```
/bin/sh ./libtool --verbose --mode=compile gcc -I./lib -I. -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -fno-strict-aliasing  -DHAVE_EXPAT_CONFIG_H -o lib/xmlparse.lo -c lib/xmlparse.c
libtool: compile:  gcc -I./lib -I. -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -fno-strict-aliasing -DHAVE_EXPAT_CONFIG_H -c lib/xmlparse.c  -fno-common -DPIC -o lib/.libs/xmlparse.o
lib/xmlparse.c:843:1: warning: 'gather_time_entropy' defined but not used [-Wunused-function]
 gather_time_entropy(void)
 ^~~~~~~~~~~~~~~~~~~
```

It's just a warning, so not the end of the world, but thought you'd like to know.

**Comment by hartwork** [comment]
Addressed by 9a1ede8d14fd13985f93a7dc4deeaccd3ae13ba8 now, thanks for testing!
